### ISSUE TITLE
not compiling

### ISSUE THREAD

**Issue by rulet** [hanzelpeter/dispmanx_vnc#29]
make
gcc -O3 main.c -o dispmanx_vncserver -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi -I/opt/vc/include/ -I/opt/vc/include/interface/vcos/pthreads -I/opt/vc/include/interface/vmcs_host/linux -L/opt/vc/lib/ -lbrcmGLESv2 -lbrcmEGL -lopenmaxil -lbcm_host -lvcos -lvchiq_arm -lpthread -lrt -lvncserver
main.c:9:10: fatal error: rfb/rfb.h: Нет такого файла или каталога
 #include <rfb/rfb.h>
          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:12: dispmanx_vncserver] Ошибка 1

**Comment by hanzelpeter** [comment]
Hello. You need to install libvncserver-dev package. This provides rfb.h files and libraries

**Comment by rulet** [comment]
Ok, compiled. How to install, sudo make install ?

**Comment by hanzelpeter** [comment]
There is no install. Just use complied binary

**Comment by rulet** [comment]
Just to run ./dispmanx_vncserver and server is up and running? Do I have to add binary to autostart for running it from boot?

**Comment by hanzelpeter** [comment]
Yes run it as sudo and it can be added for autostart manually

**Comment by rulet** [comment]
So to run it from root from boot do I have to place it in /usr/bin? What permissions and owner to set?

**Comment by hanzelpeter** [comment]
I don't known. Depends on distribution you use. I never tried to run it from boot.

**Comment by rulet** [comment]
raspbian buster

**Comment by hanzelpeter** [comment]
Raspbian has realvnc server that can be used in service mode a should do the same thing.

**Comment by rulet** [comment]
I want to control remotely kodi interface on raspberry pi. Realvnc server cannot do that with kodi(black screen). So I want to try dispmanx.

**Comment by hanzelpeter** [comment]
Have you tried Google: realvnc Enable experimental direct capture

**Comment by rulet** [comment]
Yes, tried that, doesn't work.

**Comment by hanzelpeter** [comment]
So try to place it in usr/bin and run it from some startup script. But if realvnc is not working I am no sure if dispmanx will work.

**Comment by rulet** [comment]
Is that normal behavior?

sudo ./dispmanx_vncserver rfbport 5901
[sudo] пароль для pi:
Open display[0]...
Display is 1920 x 1080
Server bpp:16
Server bigEndian:0
Server redShift:0
Server blueShift:10
Server greeShift:5
26/12/2020 15:37:10 ListenOnTCPPort: Address already in use
open /dev/uinput returned 4.
First write returned 1116.
ioctl UI_DEV_CREATE returned 0.
665/1080 Picture (015 fps) x0=0, y0=14, x1=1920, y1=1065

**Comment by hanzelpeter** [comment]
Not sure. Look like some vnc already running on 5901 port

**Comment by rulet** [comment]
That's probably realvnc, I have to stop it. Trying to connect with tiger vncviewer with 192.168.0.180::5901. It shows "unable to connect to socket ... as end computer rejected request (10061)

**Comment by rulet** [comment]
Does option about enabling vnc in raspi-config concerns about realvnc only or to just allow to for running any vnc server?

**Comment by hanzelpeter** [comment]
Should be only realvnc. I never turned vnc server on via raspi config. I always run it manually.. Try to disable reboot and then run dispmanx

**Comment by rulet** [comment]
I disabled vnc  in raspi-config, rebooted, and run:
sudo ./dispmanx_vncserver rfbport 5901
[sudo] пароль для pi:
Open display[0]...
Display is 1920 x 1080
Server bpp:16
Server bigEndian:0
Server redShift:0
Server blueShift:10
Server greeShift:5
26/12/2020 16:06:33 Listening for VNC connections on TCP port 5900
26/12/2020 16:06:33 Listening for VNC connections on TCP6 port 5900
open /dev/uinput returned 6.
First write returned 1116.
ioctl UI_DEV_CREATE returned 0.
804/1080 Picture (015 fps) x0=0, y0=14, x1=1920, y1=1065

**Comment by rulet** [comment]
Ha! Man, I changed port to 5900 and it connected, and I see kodi running tv program(but very slow). But the purpose is not to watch video but just to control kodi menu settings. But I cannot use mouse or keyboard, it does not respond.

**Comment by hanzelpeter** [comment]
Do you have X11 running and then Kodi?

**Comment by rulet** [comment]
Yes, it is regular rasbian with kodi running on it.

**Comment by hanzelpeter** [comment]
See this issue https://github.com/hanzelpeter/dispmanx_vnc/issues/27

I still did find why it doesn't work with libinput

**Comment by rulet** [comment]
Hm... Is it possible to temporaly run kodi without X11(kodi is in autostart)?

**Comment by hanzelpeter** [comment]
Don't know. Never tried Kodi.

**Comment by hanzelpeter** [comment]
Have you tried x11vnc? This is what I use when X11 is running

**Comment by rulet** [comment]
Will it destroy the system if replace xserver-xorg-input-evdev by xserver-xorg-input-libinput?

**Comment by hanzelpeter** [comment]
No. It should work. Previously there was no libinput

**Comment by rulet** [comment]
But before removing xserver-xorg-input-libinput I killed kodi process and connected again and I see only black screen. The output on raspbian during this:

sudo ./dispmanx_vncserver rfbport 5900
Open display[0]...
Display is 1920 x 1080
Server bpp:16
Server bigEndian:0
Server redShift:0
Server blueShift:10
Server greeShift:5
26/12/2020 16:58:07 Listening for VNC connections on TCP port 5900
26/12/2020 16:58:07 Listening for VNC connections on TCP6 port 5900
open /dev/uinput returned 6.
First write returned 1116.
ioctl UI_DEV_CREATE returned 0.
26/12/2020 16:58:23 Got connection from client 192.168.0.102
26/12/2020 16:58:23   other clients:
26/12/2020 16:58:23 Normal socket connection
26/12/2020 16:58:24 Client Protocol Version 3.8 y1=1080
26/12/2020 16:58:24 Protocol version sent 3.8, using 3.8
26/12/2020 16:58:24 rfbProcessClientSecurityType: executing handler for type 1
26/12/2020 16:58:24 rfbProcessClientSecurityType: returning securityResult for client rfb version >= 3.8
26/12/2020 16:58:25 Pixel format for client 192.168.0.102:
26/12/2020 16:58:25   32 bpp, depth 24, little endian
26/12/2020 16:58:25   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC6)
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0x574D5664)
26/12/2020 16:58:25 Enabling full-color cursor updates for client 192.168.0.102
26/12/2020 16:58:25 Enabling X-style cursor updates for client 192.168.0.102
26/12/2020 16:58:25 Enabling NewFBSize protocol extension for client 192.168.0.102
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFECC)
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFECD)
26/12/2020 16:58:25 Enabling LastRect protocol extension for client 192.168.0.102
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xC0A1E5CE)
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC7)
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC8)
26/12/2020 16:58:25 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEFE)
26/12/2020 16:58:25 Using compression level 2 for client 192.168.0.102
26/12/2020 16:58:25 Using image quality level 8 for client 192.168.0.102
26/12/2020 16:58:25 Using JPEG subsampling 0, Q92 for client 192.168.0.102
26/12/2020 16:58:25 Using tight encoding for client 192.168.0.102
076/1080 Picture (002 fps) x0=0, y0=0, x1=1920, y1=1080

**Comment by hanzelpeter** [comment]
Also rebooted when evdev installed?

**Comment by rulet** [comment]
No, I din't removed libinput yet. Maybe after I killed kodi process the system freezed, I don't know, I rebooted raspberry and connected again and again I see kodi running(kodi is in autostart).
I don't get.

**Comment by rulet** [comment]
So, I removed libinput... and at the end seems ui is broken -- it's some mix with gnome and lxqt for now :)). And when I started kodi it doesn't show in vnc viewer window... I'll experiment later.

**Comment by hanzelpeter** [comment]
I think libinput has dependencies. I would Google how to remove it without removing dependencies. And from now I would install a new image to raspberry pi and then only remove libinput

**Comment by hanzelpeter** [comment]
But maybe they changed something and libinput is now required

**Comment by rulet** [comment]
I found linux distribution for raspberry pi which runs just kodi, and your server just works with it as I expected(I use tigervnc viewer). Thanks for your program.
